src/api/java/com/minecolonies/api/entity/citizen/happiness/ExpirationBasedHappinessModifier.java

```diff
@@ -0,0 +1,112 @@
+package com.minecolonies.api.entity.citizen.happiness;
+
+import net.minecraft.nbt.CompoundNBT;
+
+import java.util.function.DoubleSupplier;
+
+import static com.minecolonies.api.util.constant.NbtTagConstants.TAG_DAY;
+
+/**
+ * The time based happiness modifier.
+ */
+public class ExpirationBasedHappinessModifier extends StaticHappinessModifier
+{
+    /**
+     * The number of passed days.
+     */
+    private int days = 0;
+
+    /**
+     * Period of time this modifier applies.
+     */
+    private final int period;
+
+    /**
+     * If this should give a penalty if not active.
+     */
+    private boolean inverted;
+
+    /**
+     * Create an instance of the happiness modifier.
+     *
+     * @param id     its string id.
+     * @param weight its weight.
+     * @param period the period.
+     */
+    public ExpirationBasedHappinessModifier(final String id, final double weight, final DoubleSupplier supplier, final int period)
+    {
+        super(id, weight, supplier);
+        this.period = period;
+    }
+
+    /**
+     * Create an instance of the happiness modifier.
+     *
+     * @param id     its string id.
+     * @param weight its weight.
+     * @param period the period.
+     * @param inverted if inverted.
```

Comment:
Can you crite some documentation on what inverted means? I can roughly understand it from the code, but the use case is unclear and should be documented, so that people using this method know what they are changing with this